### PR TITLE
Fix - Impacting SLA: System.ArgumentNullException

### PR DESCRIPTION
**Pull request by alishahahmed**
## Description
Fixed SLA impacting issue where customer's request without content-type header somehow made it to the service. This was resulting in a 5xx error because of the EnsureArg check in place.

I have removed this check and also updated the Handler and Reader tests such that we validate the possible input values for content-type header.

After this change, instead of 5xx status code, user should receive a 415 result code indicating that the media type is unsupported.

## Related issues
AB#94228.

## Testing
- Updated handler and receiver tests.
- All existing tests are passing.

### CODE REVIEW

File: src/Microsoft.Health.Dicom.Core/Features/Store/Entries/DicomInstanceEntryReaderManager.cs [hunk 1]
```diff
@@ -26,8 +26,6 @@ public DicomInstanceEntryReaderManager(IEnumerable<IDicomInstanceEntryReader> di
     /// <inheritdoc />
     public IDicomInstanceEntryReader FindReader(string contentType)
     {
-        EnsureArg.IsNotNull(contentType, nameof(contentType));
```
Comment (jovinson-ms) [inline src/Microsoft.Health.Dicom.Core/Features/Store/Entries/DicomInstanceEntryReaderManager.cs:29]: We should probably keep this check in place while we implement the error handling.

**Review by jovinson-ms — COMMENTED** [review]
Are we able to reproduce the conditions that would lead to this happening?

File: src/Microsoft.Health.Dicom.Core/Features/Store/Entries/DicomInstanceEntryReaderManager.cs (hunk 1 above)
Comment (wsugarman) [inline src/Microsoft.Health.Dicom.Core/Features/Store/Entries/DicomInstanceEntryReaderManager.cs:29]: Is a `ContentType` required by the spec or is there some default value we can assume? If it's the latter, then we should populate the default.

Comment (alishahahmed) [reply src/Microsoft.Health.Dicom.Core/Features/Store/Entries/DicomInstanceEntryReaderManager.cs:29]: ContentType is required by spec. No default is specified for this. In addition to the standard, we have also provided a list of accepted ContentType values in the conformance statement.

**Comment by alishahahmed** [comment]
> Are we able to reproduce the conditions that would lead to this happening?

No, we were not able to reproduce it. The controller has a Consumes attribute filter for the ContentType header. This should not have happened in the first place. I also synced with Smitha on this, and removing the EnsureArg check seemed to be the proper solution for this issue.

File: src/Microsoft.Health.Dicom.Core/Features/Store/Entries/DicomInstanceEntryReaderManager.cs (hunk 1 above)
Comment (alishahahmed) [reply src/Microsoft.Health.Dicom.Core/Features/Store/Entries/DicomInstanceEntryReaderManager.cs:29]: We don't necessarily need this. Given that the reader returns null if `ContentType` is null, we are good. The caller is responsible for making sure that the reader is not null. In our case, we always check the reader and return `UnsupportedMediaType` status code if the reader is not found.
Please do let me know if I misunderstood the comment.

Comment (alishahahmed) [reply src/Microsoft.Health.Dicom.Core/Features/Store/Entries/DicomInstanceEntryReaderManager.cs:29]: I have also updated the `StoreHandler` tests to verify that we return `UnsupportedMediaType` exception when `ContentType` is null.

**Review by jovinson-ms — APPROVED** [review]
